src/modules/keyboardmanager/ui/EditKeyboardWindow.cpp

```diff
@@ -50,6 +53,12 @@ void createEditKeyboardWindow(HINSTANCE hInst, KeyboardManagerState& keyboardMan
         MessageBox(NULL, L"Call to CreateWindow failed!", L"Error", NULL);
         return;
     }
+    // Store the newly created window's handle.
+    std::unique_lock<std::mutex> hwmdLock(editKeyboardWindowMutex);
+    hwndEditKeyboardNativeWindow = _hWndEditKeyboardWindow;
+    hwmdLock.unlock();
+
+     SetForegroundWindow(_hWndEditKeyboardWindow);
```

Comment:
I have noticed this issue before, but I don't know too much about the conventional way of implementing the functionality in C++ windows. Is this the conventional way of doing it or is there something to be handled in the window callback function to ensure it is in the foreground on creation?

Reply:
I was facing issues with the remap and shortcut window not coming on foreground when created from Settings Process that why I added a call to SetForegroundWindow but that also doesn't seems to[ work and because of ](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow)

> An application cannot force a window to the foreground while the user is working with another window. Instead, Windows flashes the taskbar button of the window to notify the user.

So I added a call to AllowSetForegroundWindow in Settings Process to allow background runner process to launch a window on foreground. More details [here](https://devblogs.microsoft.com/oldnewthing/20110613-00/?p=10443). 

I think the default behavior of window on creation is come on foreground if it can so I removed the explicit call of SetForegroundWindow after window creation it's working as expected window is created on foreground.